plugins/smartvol/init.go

```diff
@@ -0,0 +1,46 @@
+package smartvol
+
+import (
+	"github.com/gluster/glusterd2/glusterd2/servers/rest/route"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/pkg/api"
+	"github.com/gluster/glusterd2/pkg/sunrpc"
+	"github.com/gluster/glusterd2/pkg/utils"
+	smartvolapi "github.com/gluster/glusterd2/plugins/smartvol/api"
+)
+
+// Plugin is a structure which implements GlusterdPlugin interface
+type Plugin struct {
+}
+
+// Name returns name of plugin
+func (p *Plugin) Name() string {
+	return "smartvol"
+}
+
+// SunRPCProgram returns sunrpc program to register with Glusterd
+func (p *Plugin) SunRPCProgram() sunrpc.Program {
+	return nil
+}
+
+// RestRoutes returns list of REST API routes to register with Glusterd.
+func (p *Plugin) RestRoutes() route.Routes {
+	return route.Routes{
+		route.Route{
+			Name:         "SmartVolumeCreate",
+			Method:       "POST",
+			Pattern:      "/smartvol",
```

Comment:
This should not exist. The same URL should serve non-smart and smart volume create.

Reply:
For the end feature, I agree that this shouldn't be a separate endpoint. But I wouldn't mind this for the initial development and testing while we finalize the smart provisioning design.
Also, we don't yet support plugins adding middlewares yet, so it's not possible right away.

Reply:
> hundreds of users are consuming it via kubernetes provisioners

IIUC, it is probably this: `User --> KubeProvisioner --> Heketi --> Glusterd`

But are there users consuming heketi API directly ?

If kube provisioner is the primary place of heketi API consumption, the end users will not be affected if kube provisioner is modified to use new APIs.

Reply:
This is correct. It is just a few code places. Not aware of other direct consumers of the API. But the experience on the smoothness and the feature compatibility will have an impact on that large user base.